mlflow/utils/requirements_utils.py

```diff
@@ -409,3 +420,44 @@ def _get_pinned_requirement(package, version=None, module=None):
             version = version_raw
 
     return f"{package}=={version}"
+
+
+class _MismatchedPackageInfo(NamedTuple):
+    package_name: str
+    installed_version: str
+    requirement: str
+
+    def __str__(self):
+        current_status = self.installed_version if self.installed_version else "uninstalled"
+        return f"{self.package_name} (current: {current_status}, required: {self.requirement})"
+
+
+def _check_requirement_satisfied(requirement_str):
+    """
+    If installed package satisfy the requirements, return None,
+    otherwise return an instance of namedtuple `_MismatchPackageInfo`.
```

Comment:
```suggestion
    Returns None if the current python environment satisfies the given requirement. 
    Otherwise, returns an instance of `_MismatchedPackageInfo`.
```